transport/outbound.go

```diff
@@ -27,6 +27,12 @@ import "golang.org/x/net/context"
 // Outbound is a transport that knows how to send requests for procedure
 // calls.
 type Outbound interface {
+	// Sets up the outbound to start making calls.
```

Comment:
Added a note.

I'm not sure about verifying that `Start` has been called in `Call`. Seems like unnecessary synchronization overhead for not much value.